libraries/AP_Frsky_Telem/AP_Frsky_Telem.cpp

```diff
@@ -18,31 +18,55 @@
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-/* 
+/*
    FRSKY Telemetry library
 */
 
-#include "AP_Frsky_Telem.h"
-
 #include <AP_AHRS/AP_AHRS.h>
 #include <AP_BattMonitor/AP_BattMonitor.h>
 #include <AP_RangeFinder/AP_RangeFinder.h>
 #include <AP_Common/AP_FWVersion.h>
-#include <GCS_MAVLink/GCS.h>
 #include <AP_Common/Location.h>
 #include <AP_GPS/AP_GPS.h>
 #include <AP_Baro/AP_Baro.h>
+#include <AP_Param/AP_Param.h>
+#include <AP_Logger/AP_Logger.h>
+#include <AP_Vehicle/AP_Vehicle.h>
+#include <AC_Fence/AC_Fence.h>
+#include <AP_HAL/utility/sparse-endian.h>
+
+#include <stdio.h>
 #include <math.h>
 
+#include "AP_Frsky_Telem.h"
+
+#define SPORT_TX_PACKET_DUPLICATES 1
+#define SPORT_RX_PACKET_DISCARD_DUPLICATE
+
 extern const AP_HAL::HAL& hal;
 
+#if HAL_WITH_FRSKY_TELEM_BIDIRECTIONAL
+using namespace MAVLITE;
+#endif //HAL_WITH_FRSKY_TELEM_BIDIRECTIONAL
+using namespace FRSky;
+
 AP_Frsky_Telem *AP_Frsky_Telem::singleton;
 
-AP_Frsky_Telem::AP_Frsky_Telem(bool _external_data) : AP_RCTelemetry(TIME_SLOT_MAX),
-    use_external_data(_external_data)
+#if HAL_WITH_FRSKY_TELEM_BIDIRECTIONAL 
```

Comment:
I wonder if we should just assign to _frsky_parameters in the constructor itself (suitably gated with `#if`s; it will save a chunk of icky lines here.

Reply:
not sure I understand what you mean, add an argument to the constructor?